src/case-error.test.ts

```diff
@@ -2,30 +2,50 @@ import { Task, UncaughtError } from '@ts-task/task';
 import { assertFork, jestAssertNever, jestAssertUntypedNeverCalled } from './testing-utils';
 import { caseError } from './case-error';
 
-class DivisionByZeroError extends Error {
-  type = 'DivisionByZeroError';
+interface DivisionByZeroError {
+  errorType: 'DivisionByZeroError';
 }
 
-class DontLikePairsError extends Error {
-  type = 'IDontLikePairsError';
+class DontLikeEvenNumbersError extends Error {
+  type = 'DontLikeEvenNumbersError';
+
+  constructor (public evenNumber: number) {
+    super(`Ugh! ${evenNumber} is an even number!`);
+  }
 }
 
-class NewTypeOfError extends Error {
-  type = 'NewTypeOfError';
+interface NewTypeOfError {
+  type: 'NewTypeOfError';
 }
 
+const isDivisionByZeroError = <E> (err: E | DivisionByZeroError): err is DivisionByZeroError =>
```

Comment:
I think you can type err as any without any harm, the err is DivisionByZeroError is what grants it's type